python-package/lightgbm/basic.py

```diff
@@ -1115,13 +1188,144 @@ def __init__(self, data, label=None, reference=None,
         self.feature_penalty = None
         self.monotone_constraints = None
         self.version = 0
+        self._start_row = 0  # Used when pushing rows one by one.
 
     def __del__(self):
         try:
             self._free_handle()
         except AttributeError:
             pass
 
+    def create_sample_indices(self, total_nrow):
+        """Create sample indices for the given parameter of the Dataset.
+
+        Parameters
+        ----------
+        total_nrow: int
+            Total number of rows to sample from.
+            If Dataset has multiple input data, this should be the sum of rows of every file.
```

Comment:
```suggestion
            If this value is greater than the value of parameter ``bin_construct_sample_cnt``, only ``bin_construct_sample_cnt`` indices will be used.
            If Dataset has multiple input data, this should be the sum of rows of every file.
```

My first thought reading this was "what would happen if the training data has `total_nrow` is larger than the number of samples?". Looking at the changes, I think I understand the behavior. If I've understood correctly, would you please consider this documentation change?